app/models/user.rb

```diff
@@ -276,6 +281,19 @@ def reset_api_key
     self.update(api_key: Base64.encode64(md5.to_s).strip)
   end
 
+  def visible_assessments(assessment_type: nil, assessment_id: nil)
```

Comment:
This look great! Can you add a docstring to this method that describes generally what it does, what the optional arguments are and what the defaults are if the arguments aren't specified. 

We use a style of ruby docstrings that looks like:

```ruby
# Here is a description of the method below and what it does... 
# It takes three arguments +a+ +b+ and +c+ and does something with them ...
def my_function(a, b: nil, c: 5)
   .....
end

```

Reply:
I'm going to be really picky here, but I think it's worth it to be precise when writing documentation:

- change `"Defaults to Assessment"` to a description of what passing the parameter actually does. Something like: 
`If passed one of these options, only assessments of this type will be returned. Otherwise returns all assessment types`
- change `if passed and assessment...` to something like `if passed an assessment id, only the assessment with the given id will be returned if it is visible to the current user`. As it stands, it sounds like you are expecting an assessment object (not an id) and the function will return a boolean, not a collection of assessments.

Reply:
Fair enough, I agree! I've added the description of the `assessment_type` parameter and changed the description of `assessment_id` to indicate it will return a collection